File: DFS.py
def dfs(graph, root, visited):
    neighbors = graph.get(root)
    if root in visited or neighbors is None:
        return
    print(root, end=", ")
    visited.add(root)

    for neighbor in neighbors:
        dfs(graph, neighbor, visited)

#   2 - 4
#  /
# 1 - 6
#  \
#   3 - 5
graph = {
    1: [2, 3, 6],
    6: [1],
    2: [1, 4],
    3: [1, 5],
    4: [2],
    5: [3]
}

visited = set()
dfs(graph, 1, visited)

File: Java/DFS.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class DFS {

    private static void dfs(Map<Integer, List<Integer>> graph, int root, Set<Integer> visited) {

        List<Integer> neighbors = graph.get(root);
        if (visited.contains(root) || neighbors == null) {
            return;
        }
        System.out.print(root + ", ");
        visited.add(root);

        for (int neighbor : neighbors) {
            dfs(graph, neighbor, visited);
        }

    }

    /**          
     *           2 - 4
     *          /
     *         1 - 6
     *          \
     *           3 - 5
     */
    public static void main(String[] args) {
        Map<Integer, List<Integer>> graph = new HashMap<>();
        graph.put(1, Arrays.asList(2, 3, 6));
        graph.put(6, Arrays.asList(1));
        graph.put(2, Arrays.asList(1, 4));
        graph.put(3, Arrays.asList(1, 5));
        graph.put(4, Arrays.asList(2));
        graph.put(5, Arrays.asList(3));

        Set<Integer> visited = new HashSet<>();
        dfs(graph, 1, visited);
    }
    
}
